### PR TITLE
[IMP] api: Add warning about api.norecompute() deprecation

### PR DESCRIPTION
**Pull request by rousseldenis**
@rco-odoo What do you think of this ? Thanks

Since : https://github.com/odoo/odoo/commit/9920f20e4c7753bc17bea71dea3a90f7de687196#diff-6b27f146b1eec6bfa1d9d1d8b12af99e9ff01323bd7cd2841cb4fc72a30afe44R663

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/api.py [hunk 1]
```diff
@@ -821,6 +821,10 @@ def remove_to_compute(self, field, records):
     @contextmanager
     def norecompute(self):
         """ Delay recomputations (deprecated: this is not the default behavior). """
+        warnings.warn(
+            "norecompute() is deprecated. You should use protecting() instead.",
```
Comment (kmagusiak) [inline odoo/api.py:825]: Just add a version.
```suggestion
            "norecompute() is deprecated since 19.0. You should use protecting() instead.",
```

Comment (rousseldenis) [reply odoo/api.py]: @kmagusiak

**Comment by rousseldenis** [comment]
IMHO, this can be removed: https://github.com/odoo/odoo/blob/16.0/odoo/fields.py#L2381

**Review by kmagusiak — CHANGES_REQUESTED** [review]
The PR is about a deprecation. It should not reformat everything.

**Comment by rousseldenis** [comment]
> The PR is about a deprecation. It should not reformat everything.

What do you mean ?

**Comment by kmagusiak** [comment]
> > The PR is about a deprecation. It should not reformat everything.
> 
> What do you mean ?

The diff reformats now all the file.

**Comment by rousseldenis** [comment]
@kmagusiak Done. What do you think about removing the norecompute call in `Binary` field class ? It does nothing.

**Comment by kmagusiak** [comment]
You will have to fix the runbot, so I'm not sure this should target v16. @ryv-odoo what do you think?

**Comment by rousseldenis** [comment]
> You will have to fix the runbot, so I'm not sure this should target v16. @ryv-odoo what do you think?

In fact, removing the call to `norecompute()` won't harm anything (that is done in further versions) as it does nothing:

https://github.com/odoo/odoo/blob/9bb45ec35ae34f0c9de320072650bbb63f23d543/odoo/fields.py#L2381

**Comment by ryv-odoo** [comment]
Hello @rousseldenis , 

What is the purpose of this change (why do you want to backport https://github.com/odoo/odoo/pull/99550) ? We always deprecated and cleaned the master version only, never the stable ones. You can check our stable policy [here](https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean). In this case, you may potentially create new warning logs on stable servers that may trigger some stressful situations for the server managers (for no reason).

I hope you understand. 
Sincerely.

**Pull request closed without merging**